pkg/controllers/user/logging/deployer/appdeployer.go

```diff
@@ -141,11 +142,40 @@ func (d *AppDeployer) isDeploySuccess(targetNamespace string, selector map[strin
 	}
 }
 
-func rancherLoggingApp(appCreator, systemProjectID, catalogID, driverDir, dockerRoot string) *projectv3.App {
+func rancherLoggingApp(appCreator, systemProjectID, catalogID, driverDir, dockerRoot string, tolerations *[]corev1.Toleration) *projectv3.App {
 	appName := loggingconfig.AppName
 	namepspace := loggingconfig.LoggingNamespace
 	_, systemProjectName := ref.Parse(systemProjectID)
 
+	answers := map[string]string{
+		//compatible with old version
+		"fluentd.enabled":              "true",
+		"fluentd.cluster.dockerRoot":   dockerRoot,
+		"log-aggregator.enabled":       "true",
+		"log-aggregator.flexVolumeDir": driverDir,
+
+		//new version
+		"fluentd.fluentd-linux.enabled":                     "true",
+		"log-aggregator.log-aggregator-linux.enabled":       "true",
+		"log-aggregator.log-aggregator-linux.flexVolumeDir": driverDir,
+	}
+
+	if tolerations != nil {
+		for i, toleration := range *tolerations {
+			// linux
+			answers[fmt.Sprintf("log-aggregator.log-aggregator-linux.tolerations[%d].key", i)] = toleration.Key
+			answers[fmt.Sprintf("log-aggregator.log-aggregator-linux.tolerations[%d].value", i)] = toleration.Value
+			answers[fmt.Sprintf("log-aggregator.log-aggregator-linux.tolerations[%d].effect", i)] = string(toleration.Effect)
+			answers[fmt.Sprintf("log-aggregator.log-aggregator-linux.tolerations[%d].operator", i)] = string(toleration.Operator)
```

Comment:
This will not fix the problem but allows users to configure tolerations in a way that schedules them on the control plane and etcd nodes.  This will require a UI change.